src/node_task_runner.cc

```diff
@@ -256,20 +267,19 @@ void RunTask(std::shared_ptr<InitializationResultImpl> result,
 // If the "--" flag is not found, it returns an empty optional.
 // Otherwise, it returns the positional arguments as a single string.
 // Example: "node -- script.js arg1 arg2" returns "arg1 arg2".
-std::optional<std::string> GetPositionalArgs(
-    const std::vector<std::string>& args) {
+PositionalArgs GetPositionalArgs(const std::vector<std::string>& args) {
   // If the "--" flag is not found, return an empty optional
   // Otherwise, return the positional arguments as a single string
   if (auto dash_dash = std::find(args.begin(), args.end(), "--");
       dash_dash != args.end()) {
-    std::string positional_args;
+    PositionalArgs positional_args{};
     for (auto it = dash_dash + 1; it != args.end(); ++it) {
-      positional_args += it->c_str();
+      positional_args.push_back(std::string_view(it->c_str(), it->size()));
```

Comment:
You may add a comment here to the effect that positional_args which is returned by GetPositionalArgs, contains string views that point at args, that is being passed to GetPositionalArgs. It is a life cycle issue that can be tricky for the reader to sort out in the abstract.

(I expect that the code is fine, I just recommend a comment, a single line would do.)

Reply:
Definitely. I've updated the pull-request.